src/Tools/dotnet-monitor/Auth/DeferredAuthenticationValidationException.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Auth
+{
+    internal sealed class DeferredAuthenticationValidationException : Exception
+    {
+        public string ConfigurationPath { get; }
+        public IEnumerable<string> FailureMessages { get; }
+
+        public DeferredAuthenticationValidationException(string configurationPath, IEnumerable<ValidationResult> results) : base()
+        {
+            ConfigurationPath = configurationPath;
+
+            List<string> failureMessages = new (results.Count());
+            foreach (ValidationResult result in results)
+            {
+                if (result.ErrorMessage != null)
```

Comment:
nit: not sure if applicable in this use case, but should this use `IsEmptyOrNull`?

Reply:
Agreed, no point in adding a failure message that is empty.

Reply:
Maybe even better to use `IsNullOrWhitespace`.